tools/compare_benchmark.py

```diff
@@ -0,0 +1,159 @@
+# Runs nvfuser_bench on two commits and compares their results. See
+# https://github.com/NVIDIA/Fuser/wiki/Developer-guide#benchmark-nvfuser for
+# usage.
+
+import argparse
+from collections.abc import Iterable
+from dataclasses import dataclass
+import json
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import subprocess
+
+
+# Runs nvfuser_bench with `benchmark_args` on the given branch or commit. Dumps
+# outputs to `out_dir`. Returns `out_dir`/`branch_or_commit`.json that captures
+# the benchmark result. If the output already exists, skips benchmarking and
+# uses that output. This is useful, for example, when comparing multiple
+# contenders to the same base.
+def run_benchmark(branch_or_commit: str, benchmark_args: str, out_dir: str) -> str:
+    benchmark_out = os.path.join(out_dir, branch_or_commit + ".json")
+    if os.path.exists(benchmark_out):
+        print(f"{benchmark_out} already exists. Skip benchmarking {branch_or_commit}.")
+        return benchmark_out
+
+    subprocess.check_call(f"git checkout {branch_or_commit}", shell=True)
+
+    subprocess.check_call("pip install -e .", shell=True)
+
+    benchmark_command = " ".join(
+        (
+            "bin/nvfuser_bench",
+            benchmark_args,
+            f"--benchmark_out={benchmark_out}",
+            "--benchmark_format=json",
+        )
+    )
+    print("Running benchmark command: " + benchmark_command)
+    with open(os.path.join(out_dir, branch_or_commit + ".stdout"), "w") as stdout, open(
+        os.path.join(out_dir, branch_or_commit + ".stderr"), "w"
+    ) as stderr:
+        subprocess.check_call(
+            benchmark_command, stdout=stdout, stderr=stderr, shell=True
+        )
+    print(f"The benchmark output is stored in {benchmark_out}.")
+
+    return benchmark_out
+
+
+@dataclass
+class Comparison:
+    name: str
+    baseline_time: float
+    contender_time: float
+    # contender_time divided by baseline_time. Smaller is better.
+    change: float
```

Comment:
You could compute this in `__post_init__`.

Reply:
Yes, I could. Currently, I'm reading this value from the `.json` file, which is easier than computing it.